pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -0,0 +1,473 @@
+        return te_flags_passed
+
+    def select_precision_plugin(self):
+        if self.precision == 32:
+            self.amp_type = None
+            return PrecisionPlugin()
+
+        elif self.precision == 16:
+            if self.on_tpu:
+                return TPUHalfPrecisionPlugin()
+
+            if self.amp_type == "native":
+                if not _NATIVE_AMP_AVAILABLE:
+                    rank_zero_warn(
+                        "You have asked for native AMP but your PyTorch version does not support it."
+                        " Consider upgrading with `pip install torch>=1.6`."
+                        " We will attempt to use NVIDIA Apex for this session."
+                    )
+                    self.amp_type = "apex"
+                else:
+                    log.info("Using native 16bit precision.")
+                    if self.distributed_backend == "ddp_sharded" or self.distributed_backend == "ddp_sharded_spawn":
+                        return ShardedNativeMixedPrecisionPlugin()
+                    self.amp_type = AMPType.NATIVE
+                    return NativeMixedPrecisionPlugin()
+
+            if self.amp_type == "apex":
+                if not _APEX_AVAILABLE:
+                    rank_zero_warn(
+                        "You have asked for Apex AMP but you have not installed it yet."
+                        " Install apex first using this guide: https://github.com/NVIDIA/apex#linux"
+                    )
+                else:
+                    if self.distributed_backend == "ddp_sharded" or self.distributed_backend == "ddp_sharded_spawn":
+                        raise MisconfigurationException(
+                            "Sharded Plugin is not supported with Apex AMP, "
+                            "please using native AMP for 16-bit precision."
+                        )
+                    log.info("Using APEX 16bit precision.")
+                    self.amp_type = AMPType.APEX
+                    return ApexMixedPrecisionPlugin(self.amp_level)
+        else:
+            raise NotImplementedError("We only support precisions 32 and 16!")
+
+    def select_training_type_plugin(self):
+        cluster_environment = self.select_cluster_environment()
+        if self.use_ddp2:
+            plugin = DDP2Plugin(parallel_devices=self.parallel_devices, cluster_environment=cluster_environment)
+        elif self.use_ddp:
+            use_slurm_ddp = self.use_ddp and self.is_slurm_managing_tasks
+            use_torchelastic_ddp = self.use_ddp and self.is_using_torchelastic
+            use_ddp_spawn = self._distrib_type == DistributedType.DDP_SPAWN
+            use_ddp_cpu_spawn = self.use_ddp and self.on_cpu
+            use_ddp_cpu_torch_elastic = use_ddp_cpu_spawn and self.is_using_torchelastic
+            use_ddp_cpu_slurm = use_ddp_cpu_spawn and self.is_slurm_managing_tasks
+            # use_ddp_sharded = self.distributed_backend == "ddp_sharded"
```

Comment:
Does sharded still work ?

Reply:
yes but deactivated to avoid failing tests until we have the classes merged